modules/tinymce/src/core/main/ts/api/util/I18n.ts

```diff
@@ -102,8 +116,8 @@ const translate = (text: Untranslated): TranslatedString => {
 
   const getLangData = (text: Untranslated) => {
     // make sure we work on a string and return a string
-    const textstr = toString(text);
-    return Obj.get(langData, textstr.toLowerCase()).map(toString).getOr(textstr);
+    const textStr = toString(text);
+    return Obj.get(langData, Obj.has(langData, textStr) ? textStr : textStr.toLowerCase()).map(toString).getOr(textStr);
```

Comment:
Can we have `Obj.has(langData, textStr) ? textStr : textStr.toLowerCase()` as separate variable above. Seems like the line is getting bit long  

This is another alternative that might be a bit clearer

```suggestion
    return Obj.get(langData, textStr)
    	.orThunk(() => Obj.get(langData, textStr.toLowerCase())
    	.map(toString)
    	.getOr(textStr);
```

Reply:
The goal of rewriting this was to remove duplicated calls to `Obj.has` which happens as part of `Obj.get` to improve performance. I've have a couple of ideas doing it: https://github.com/tinymce/tinymce/pull/8995#discussion_r1305360182

Reply:
Thunk was one of the options I considered, but we generate enough short-lived objects as it is and this code will run close to 1000 times with all the plugins loaded